lucene/facet/src/test/org/apache/lucene/facet/taxonomy/directory/TestDirectoryTaxonomyReader.java

```diff
@@ -569,4 +569,32 @@ public void testAccountable() throws Exception {
     taxoReader.close();
     dir.close();
   }
+
+  public void testCallingBulkPathReturnsCorrectResult() throws Exception {
+    Directory src = newDirectory();
+    DirectoryTaxonomyWriter w = new DirectoryTaxonomyWriter(src);
+    String randomArray[] = new String[random().nextInt(1000)];
+    Arrays.setAll(randomArray, i -> Integer.toString(random().nextInt()));
+
+    FacetLabel allPaths[] = new FacetLabel[randomArray.length];
+    int allOrdinals[] = new int[randomArray.length];
+
+    for (int i = 0; i < randomArray.length; i++) {
+      allPaths[i] = new FacetLabel(randomArray[i]);
+      w.addCategory(allPaths[i]);
```

Comment:
Could we randomly `w.commit()` in here with low but non-zero chance?  This way we will sometimes exercise a multi-segment taxo index.

And, could you temporarily put the `assert` back from the first revision and confirm this test case eventually would catch that bug?

Reply:
Done. 

Thanks for the suggestion. I verified that this code trips when `values.advanceExact` is removed, **but** I was not able to verify this by just keeping the `assert values.advanceExact(ordinals[i] - leafReaderDocBase)` line. Perhaps assert was always being executed?

I also manually verified (with print statements) that multi segments were created and were working as expected.

Reply:
OK, thanks for confirming.

Hmm, that is spooky if `assert` is always enabled for Lucene tests!  We may have lost this functionality in the gradle migration?  Maybe, separately confirm this with a tiny test case, and if you cannot get that tiny test case to fail, send an email to `dev@` list asking about it?

Reply:
I got a small test case working that modified a global variable 1000 times (via an `assert`) and checked the final value. It seems that `assert` is always enabled. 

I'll send an email about the same